app/src/main/kotlin/com/x8bit/bitwarden/ui/vault/feature/vault/util/VaultDataExtensions.kt

```diff
@@ -233,6 +282,7 @@ private fun CipherListView.toVaultItemOrNull(
             extraIconList = toLabelIcons(),
             shouldShowMasterPasswordReprompt = hasMasterPassword &&
                 reprompt == CipherRepromptType.PASSWORD,
+            hasDecryptionError = hasDecryptionError,
```

Comment:
Can we use `hasDecryptionError` to control the overflow options that are assigned? That would make the `item.hasDecryptionError` checks in `VaultContent` unnecessary. Something like...

```kotlin
overflowOptions = toOverflowOptions(...).takeUnless { hasDecryptionError }
```

or if we don't want `toOverflowOptions()` to execute...

```kotlin
overflowOptions = if (hasDecryptionError) {
    emptyList()
} else { 
    toOverflowOptions()
}
```

and in `VaultContent` we can revert
```kotlin
overflowOptions = if (favoriteItem.hasDecryptionError) {
    persistentListOf()
} else {
    favoriteItem.overflowOptions.toImmutableList()
},
```

Reply:
I think we still need the decryption error in the content to know at the click event if the cipher has an error and handle it accordingly